cluster-autoscaler/vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -595,7 +595,7 @@ type CloudConfig struct {
 	}
 }
 
-func (cfg *CloudConfig) validateOverrides() error {
+func (cfg *CloudConfig) ValidateOverrides() error {
```

Comment:
Were all the methods capitalized by syncing the vendored file or by hand?

Reply:
Yes, that is how I did it to get the change in here quickly.

Reply:
Anything under vendor dir is untouchable. Since anytime autoscaler dependencies are updated, your changes are lost. The original idea was to have every component to implement its own mechanism of overriding the custom endpoints. Once we update all relevant component, we may identify which code is generic enough to be shared among components. Right now, it means to copy paste both `validateOverrides` and `getResolver` methods from the vendor dir into `cluster-autoscaler/cloudprovider/aws/aws_manager.go`.

Reply:
Thanks for the input. I replicated the necessary changes from vendor into cluster-autoscaler/cloudprovider/aws/aws_manager.go: CloudConfig struct and validateOverrides and getResolver methods.
Should autoscaler have it's own AWS config file with overrides and not rely on the one necessary for the Openshift install? I assume so to adhere to the original idea of every component implementing it's own mechanism.

Reply:
The intention (as I understand it) is to replicate the same code in here. Share the config struct, but just have another implementation of it here. That's how these are set up for the time being and have behaved historically.

Reply:
Ok, so I'll change it to reuse the vendor provided CloudConfig struct. I thought the intent may have been to completely remove any dependency on the vendor code.

Reply:
> Ok, so I'll change it to reuse the vendor provided CloudConfig struct. I thought the intent may have been to completely remove any dependency on the vendor code.

AWS endpoint customization was merged only into kubernetes master branch. New feature are not usually backported into older releases unless they are proven very useful and important. Given (and please prove me wrong) the autoscaler deps are synced only with the stable versions of kubernetes, it would take some time to fetch the master changes into the vendor directory anyway. So yes, there was no intent to have the AWS endpoint customization to be consumed from the vendor directory.